Add option to collapse archived files

AndreiVernon opened the pull request:
The main purpose of this feature is to make memo compatible with the mpv manga reader script.

In its current state, memo will add an entry to the history log for every single page in a manga file, and when you try to open one from memo's menu, it will open only that page instead of the entire manga file.

When enabled, memo will only show the most recent entry for any archive. If you open one of these entries from memo's history menu, it will tell mpv to load the path to the archive, not the path to any specific file within the archive. This also allows mpv to auto-resume the manga file from where the reader left off.

This feature only changes the show_history code, so every page is still added to the history log.

AndreiVernon committed:
add archive collapsing code

AndreiVernon committed:
Update memo.conf

AndreiVernon committed:
Update memo.conf

AndreiVernon committed:
Update memo.lua

AndreiVernon committed:
fix archive display

po5 commented:
Thanks for the contribution,

I disagree that it's memo's job to solve the playlist issue. This is useful to more than just memo users.
I wrote a script [autoload-archive-hook.lua](https://gist.github.com/po5/2415cb39c94760ac8c2c7666a30bf02c) that fixes this issue, let me know if it works well for you.

I will welcome a feature that keeps only the latest entry for any given archive, but it has to:
- Display and point to the inner file (no redirecting to the archive)
- Integrate with the path_info system, the order of its parsing logic is deliberate and very important, and we must not do extra parsing outside of it
- Not override the `loadfile` like in the current state of this PR

Name it something like `hide_same_archive` to be in line with `hide_same_dir`.

AndreiVernon committed:
Redo archive detection and basename logic

Use path_info parsing for archive protocol and archive basename.
Collapse entries by making files in the same archive have the same cache_key.
Fix path_info bug pertaining to magnet links.
Add separate option for archive names replacing inner file names.

AndreiVernon committed:
Update memo.conf to match script changes

AndreiVernon committed:
Improve archive handling

Use known_archives instead of known_files, so as to not bypass other show_history checks.
display_archive_name: Set display_path to archive path instead of manually finding archive basename.

AndreiVernon committed:
Finalize display_archive_name

fix title string clear

AndreiVernon committed:
clarify config file

AndreiVernon committed:
fix archive display name

properly implements `archive: inner file` format when display_archive_name is false
